component/kafka/group/option.go

```diff
@@ -26,6 +29,52 @@ func FailureStrategy(fs kafka.FailStrategy) OptionFunc {
 	}
 }
 
+// CheckTopic will attempt to:
+//	1. connect to the broker,
+//  2. retrieve the existing topics in the broker
+//  3. if auto-topic creation is disabled, check whether the configured topics exist in the broker
+// If any of the checks above fail the component will exit before starting to consume messages
+func CheckTopic() OptionFunc {
+	return func(c *Component) error {
+		c.topicChecker = func(ctx context.Context, c *Component) error {
+			client, err := sarama.NewClient(c.brokers, c.saramaConfig)
+			defer func() {
```

Comment:
The defer should rather be called after `if err != nil`, this way, we don't need to check whether `client` is nil or not.